src/workarounds/forwarddiff_rules.jl

```diff
@@ -208,9 +208,9 @@ function self_consistent_field(basis_dual::PlaneWaveBasis{T};
         ρ_dual = DFTK.compute_density(basis_dual, ψ_dual, occupation_dual)
         εF_dual = T(scfres.εF)  # Only needed for entropy term
         eigenvalues_dual = [T.(εk) for εk in scfres.eigenvalues]
-        _, ham_dual = energy_hamiltonian(basis_dual, ψ_dual, occupation_dual;
-                                         ρ=ρ_dual, eigenvalues=eigenvalues_dual, 
-                                         εF=εF_dual)
+        ham_dual = energy_hamiltonian(basis_dual, ψ_dual, occupation_dual;
+                                      ρ=ρ_dual, eigenvalues=eigenvalues_dual,
+                                      εF=εF_dual).H
```

Comment:
Hmm. What I dislike about this rewrite is that I have to go to the very end (and almost miss) that this is only taking the `H` part. We can solve this in Julia 1.7 by using the `(; H) = ` unpacking, but for now maybe we should make an exception for this type of unpacking?

Reply:
no you are right. You cannot rename things in that way. So in this case we would need the `.ham` prefix as well.

Ok. To be honest I don't know. The prefix is not so great, but `_, ham = ` is not awesome either. My tendency is to leave it as it was, but I'm happy to be overruled here.